pkg/controller/clustercontroller/backups.go

```diff
@@ -0,0 +1,172 @@
+	myclientset "github.com/presslabs/mysql-operator/pkg/generated/clientset/versioned"
+)
+
+var (
+	lockJobRegister sync.Mutex
+	// polling time for backup to be completed
+	backupPollingTime = time.Second
+	// time to wait for a backup to be completed
+	backupWatchTimeout = time.Hour
+)
+
+// The job structure contains the context to schedule a backup
+type job struct {
+	Name      string
+	Namespace string
+
+	BackupRunning *bool
+
+	lock     *sync.Mutex
+	myClient myclientset.Interface
+}
+
+func (c *Controller) registerClusterInBackupCron(cluster *api.MysqlCluster) error {
+	glog.Infof("Register cluster into cronjob: %s, crontab: %s",
+		cluster.Name, cluster.Spec.BackupSchedule)
+
+	if len(cluster.Spec.BackupSchedule) == 0 {
+		return nil
+	}
+
+	schedule, err := cron.Parse(cluster.Spec.BackupSchedule)
+	if err != nil {
+		return fmt.Errorf("fail to parse schedule: %s", err)
+	}
+
+	lockJobRegister.Lock()
+	defer lockJobRegister.Unlock()
+
+	for _, entry := range c.cron.Entries() {
+		j, ok := entry.Job.(job)
+		if ok && j.Name == cluster.Name && j.Namespace == cluster.Namespace {
+			glog.V(3).Infof("Cluster %s already added to cron.", cluster.Name)
+			if !reflect.DeepEqual(entry.Schedule, schedule) {
+				glog.Infof("Update cluster '%s' scheduler to: %s",
+					cluster.Name, cluster.Spec.BackupSchedule)
+				c.cron.Remove(cluster.Name)
+				break
+			}
+			return nil
+		}
+	}
+
+	c.cron.Schedule(schedule, job{
+		Name:          cluster.Name,
+		Namespace:     cluster.Namespace,
+		myClient:      c.myClient,
+		BackupRunning: new(bool),
+		lock:          new(sync.Mutex),
+	}, cluster.Name)
+
+	return nil
+}
+
+func (j job) Run() {
+	backupName := fmt.Sprintf("%s-auto-backup-%s", j.Name, time.Now().Format("2006-01-02t15-04-05"))
+	glog.Infof("Schedul backup job started. Creating backup %s..", backupName)
+
+	// Wrap backup creation to ensure that lock is released when backup is
+	// created
+	created := func() bool {
+		j.lock.Lock()
+		defer j.lock.Unlock()
+
+		if *j.BackupRunning {
+			glog.Infof("Last scheduled backup still running! Can't initiate a new backup for cluster: %s",
+				j.Name)
+			return false
+		}
+
+		tries := 0
+		for {
+			var err error
+			_, err = j.myClient.Mysql().MysqlBackups(j.Namespace).Create(&api.MysqlBackup{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: backupName,
+					Labels: map[string]string{
+						"recurrent": "true",
+					},
+				},
+				Spec: api.BackupSpec{
+					ClusterName: j.Name,
+				},
+			})
+
+			if err == nil {
+				break
+			}
+			glog.V(3).Infof("Fail to create backup %s, err: %s", backupName, err)
+
+			if tries > 5 {
+				glog.Errorf("Fail to create backup for cluster: %s, err: %s, max tries %d exeded!",
+					j.Name, err, tries)
+				return false
+			}
+
+			time.Sleep(5 * time.Second)
+			tries += 1
+		}
+
+		*j.BackupRunning = true
+		return true
+	}()
+
+	if !created {
+		return
+	}
+
+	defer func() {
+		j.lock.Lock()
+		defer j.lock.Unlock()
+		*j.BackupRunning = false
```

Comment:
I think that by keeping track of individual backups we could control how many backups might end up running in parallel and it would also allow us to Poll the results indefinitely, in a separate function. That being said, the current implementation is good enough.